android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt

```diff
@@ -258,92 +264,51 @@ fun Array<ActionParameter>?.toParamDataMap(): Map<String, String> =
 
-fun List<OverflowMenuItemConfig>.decodeBinaryResourcesToBitmap(
-  coroutineScope: CoroutineScope,
-  registerRepository: RegisterRepository,
-  decodedImageMap: SnapshotStateMap<String, Bitmap>,
-) {
-  this.forEach {
-    val resourceId = it.icon!!.reference!!.extractLogicalIdUuid()
-    coroutineScope.launch() {
-      registerRepository.loadResource<Binary>(resourceId)?.let { binary ->
-        decodedImageMap[resourceId] = binary.data.decodeToBitmap()
-      }
-    }
-  }
-}
-
-fun Sequence<NavigationMenuConfig>.decodeBinaryResourcesToBitmap(
-  coroutineScope: CoroutineScope,
-  registerRepository: RegisterRepository,
+suspend fun Sequence<String>.resourceReferenceToBitMap(
+  fhirEngine: FhirEngine,
   decodedImageMap: SnapshotStateMap<String, Bitmap>,
 ) {
-  this.forEach {
-    val resourceId = it.menuIconConfig!!.reference!!.extractLogicalIdUuid()
-    coroutineScope.launch() {
-      registerRepository.loadResource<Binary>(resourceId)?.let { binary ->
-        decodedImageMap[resourceId] = binary.data.decodeToBitmap()
-      }
+  forEach {
+    val resourceId = it.extractLogicalIdUuid()
+    fhirEngine.loadResource<Binary>(resourceId)?.let { binary ->
+      decodedImageMap[resourceId] = binary.data.decodeToBitmap()
     }
   }
 }
 
-suspend fun loadRemoteImagesBitmaps(
-  views: List<ViewProperties>,
-  registerRepository: RegisterRepository,
-  computedValuesMap: Map<String, Any>,
+suspend fun List<ViewProperties>.decodeImageResourcesToBitmap(
+  fhirEngine: FhirEngine,
   decodedImageMap: MutableMap<String, Bitmap>,
 ) {
-  suspend fun ViewProperties.loadIcons() {
-    when (this.viewType) {
+  val queue = ArrayDeque(this)
+  while (queue.isNotEmpty()) {
+    val viewProperty = queue.removeFirst()
+    when (viewProperty.viewType) {
       ViewType.IMAGE -> {
-        val imageProps = this as ImageProperties
-        if (
-          !imageProps.imageConfig?.reference.isNullOrEmpty() &&
-            imageProps.imageConfig?.type == ICON_TYPE_REMOTE
-        ) {
-          try {
-            val resourceId =
-              imageProps.imageConfig
-                ?.reference
-                ?.interpolate(computedValuesMap)
-                ?.extractLogicalIdUuid()
-
-            if (resourceId != null) {
-              registerRepository.loadResource<Binary>(resourceId)?.let { binary ->
-                decodedImageMap[resourceId] = binary.data.decodeToBitmap()
-              }
+        val imageProperties = (viewProperty as ImageProperties)
+        if (imageProperties.imageConfig != null) {
+          val imageConfig = imageProperties.imageConfig
+          if (
+            ICON_TYPE_REMOTE.equals(imageConfig?.type, ignoreCase = true) &&
+              !imageConfig?.reference.isNullOrBlank()
+          ) {
+            val resourceId = imageConfig!!.reference!!
+            fhirEngine.loadResource<Binary>(resourceId)?.let { binary: Binary ->
+              decodedImageMap[resourceId] = binary.data.decodeToBitmap()
```

Comment:
I suggest using a try-catch block in this situation. 
The reason for this is that the data might not always be valid or properly formatted, which could lead to exceptions being thrown. 
This might help prevent potential app crashes.

Reply:
The `fhirEngine.loadResource` function has a try-catch block that returns null if an exception is caught. Or did you mean catching exceptions thrown while decoding to bitmap?